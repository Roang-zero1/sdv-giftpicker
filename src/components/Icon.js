import React, { Component } from 'react';
import images from '../data/GiftImages';
import styled from 'styled-components';

const Img = styled.img`
  left: -12px;
  position: relative;
  max-width: 24px;
`;

class Icon extends Component {
  render() {
    return <Img className="icon" src={images[this.props.gift]} alt="" />;
  }
}

export default Icon;
