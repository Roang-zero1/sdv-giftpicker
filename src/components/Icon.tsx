import * as React from 'react';
import { Component } from 'react';
import styled from 'styled-components';

import images from '../data/GiftImages';

export interface IProps {
  gift: number;
  grayscale?: boolean;
}

interface IImgProps {
  grayscale?: boolean;
}

const Img = styled<IImgProps, 'img'>('img')`
  max-width: 24px;
  filter: ${props => (props.grayscale ? 'grayscale(100%)' : undefined)};
`;

class Icon extends Component<IProps> {
  public render() {
    const { gift, grayscale } = this.props;
    return <Img grayscale={grayscale} src={images[gift]} alt="" />;
  }
}

export default Icon;
