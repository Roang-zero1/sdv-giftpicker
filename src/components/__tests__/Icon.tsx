import 'jest-styled-components';

import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';

import toJson from 'enzyme-to-json';
import * as React from 'react';
import Icon, { IProps } from '../Icon';

const gift = 22;

describe('components/Icon --- Shallow render component', () => {
  let cut: ShallowWrapper;
  let props: IProps;

  beforeEach(() => {
    props = { gift };
    cut = shallow(<Icon {...props} />);
  });

  it('should render the one component', () => {
    expect(cut.length).toEqual(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(cut).toMatchSnapshot();
  });
});

describe('components/Icon --- Render component', () => {
  let renderedCut: ReactWrapper | undefined;
  let props: IProps;

  const cut = () => {
    if (!renderedCut) {
      renderedCut = mount(<Icon {...props} />);
    }
    return renderedCut;
  };

  beforeEach(() => {
    renderedCut = undefined;
    props = { gift };
  });

  it('should render the one component', () => {
    expect(cut().length).toEqual(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(cut()).toMatchSnapshot();
  });

  it('should gray out icons when grayscale is applied', () => {
    props = { ...props, grayscale: true };
    expect(toJson(cut().find('img'))).toHaveStyleRule(
      'filter',
      'grayscale(100%)'
    );
  });
});
