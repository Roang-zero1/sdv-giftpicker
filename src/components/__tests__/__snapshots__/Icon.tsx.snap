// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Icon --- Render component should be the same as the last snapshot 1`] = `
.c0 {
  height: auto;
  max-width: 24px;
}

<GiftIcon
  gift={22}
>
  <styled.img
    alt=""
    src="Dandelion.png"
  >
    <img
      alt=""
      className="c0"
      src="Dandelion.png"
    />
  </styled.img>
</GiftIcon>
`;

exports[`components/Icon --- Shallow render component should be the same as the last snapshot 1`] = `
<styled.img
  alt=""
  src="Dandelion.png"
/>
`;
